corehq/apps/userreports/views.py

```diff
@@ -861,6 +860,52 @@ def report_source_json(request, domain, report_id):
     return json_response(config)
 
 
+class ExpressionDebuggerView(BaseUserConfigReportsView):
+    urlname = 'expression_debugger'
+    template_name = 'userreports/expression_debugger.html'
+    page_title = ugettext_lazy("Expression Debugger")
+
+
+@login_and_domain_required
+@toggles.USER_CONFIGURABLE_REPORTS.required_decorator()
+def evaluate_expression(request, domain):
+    doc_type = request.POST['doc_type']
```

Comment:
interesting. i think that makes sense but not going to prioritize now unless you think important. generally you always know whether you want a form or case since that's how data sources work so I think it's not too painful for users (and possibly might save someone debugging time if they don't realize they pasted the wrong ID or something).